### PR TITLE
bump: upgrade version of go to 1.18

### PR DESCRIPTION
**Pull request by openwithcode**
## Description

We are upgrading go to version 1.18 in order to take advantage of the generics functionality that was added.

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -14,7 +14,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-FROM golang:1.17.8-buster AS golang-build
+FROM golang:1.8.0 AS golang-build
```
Comment (myungjin) [inline build/Dockerfile:17]: Is this correct?

**Review by myungjin — COMMENTED** [review]
left one comment. Can you also confirm that the upgrade didn't cause any issue?

File: build/Dockerfile (hunk 1 above)
Comment (openwithcode) [reply build/Dockerfile]: I've change it to `golang:1.18.0`  which exists

**Review by myungjin — APPROVED** [review]
lgtm

**Pull request merged**